### ISSUE TITLE
Support MARK inside /* */ block (at least in JS/TS)

### ISSUE THREAD

**Issue by oriSomething** [callodacity/comment-anchors#11]
I've find many cases when I have something like the next block:
```js
/**
 * Helpers
 * ===
 * NOTE: BLAH BLAH
 */
//// Helpers
```

Which is quite ugly I think possible solution is something like:

```js
/**
 * MARK: Helpers
 * ===
 * NOTE: BLAH BLAH
 */
```
Of course it means some serious change to the internal. I'm not sure but maybe atom support telling us what is the scope of some part of the code so we will be able to smarter check. Even though I can take the risk it will add marks inside template literal.

WDYT?

**Comment by callodacity** [comment]
Originally this was a simple regex based search over the text editor. Would adding your own custom regex in the package settings solve your issue?

If not, I can see the benefit of a more sophisticated search method. I think implementing a quick check of the current scope before creating an anchor is important though, we could use [`::scopeDescriptorForBufferPosition`](https://atom.io/docs/api/v1.18.0/TextEditor#instance-scopeDescriptorForBufferPosition) in the `textEditor` API.

**Comment by oriSomething** [comment]
I prefer not adding custom option just yet. Because moving for using the scope of the buffer will be safer move before this, which will make it easier to extend the  support and by default support more languages.
For example if we know we're in comment, using regex such as `^\sMARK:` when `^` means more of the beginning of the scope will help to support any languages with `MARK:` by default

**Comment by callodacity** [comment]
Yeah that's definitely a better approach, I say we go with that and try it out.
